File: GXpert/GXpert.Web/Modules/ServerTypes/Attendance/PollResponseForm.ts
import { IntegerEditor, StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface PollResponseForm {
    PollId: IntegerEditor;
    StudentId: IntegerEditor;
    Response: StringEditor;
    ResponseTimeInSeconds: IntegerEditor;
    Score: IntegerEditor;
    LiveSessionLogId: IntegerEditor;
    ActivationId: IntegerEditor;
}

export class PollResponseForm extends PrefixedContext {
    static readonly formKey = 'Attendance.PollResponse';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PollResponseForm.init)  {
            PollResponseForm.init = true;

            var w0 = IntegerEditor;
            var w1 = StringEditor;

            initFormType(PollResponseForm, [
                'PollId', w0,
                'StudentId', w0,
                'Response', w1,
                'ResponseTimeInSeconds', w0,
                'Score', w0,
                'LiveSessionLogId', w0,
                'ActivationId', w0
            ]);
        }
    }
}

File: GXpert/GXpert.Web/Modules/Attendance/PollResponse/PollResponseForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Attendance.Forms;

[FormScript("Attendance.PollResponse")]
[BasedOnRow(typeof(PollResponseRow), CheckNames = true)]
public class PollResponseForm
{
    [HalfWidth]
    public int PollId { get; set; }
    [HalfWidth]
    public int StudentId { get; set; }
    [HalfWidth]
    public string Response { get; set; }
    [HalfWidth]
    public int ResponseTimeInSeconds { get; set; }
    [HalfWidth]
    public int Score { get; set; }
    [HalfWidth]
    public int LiveSessionLogId { get; set; }
    [HalfWidth]
    public int ActivationId { get; set; }
}